src/api_manager/check_service_control.cc

```diff
@@ -31,38 +31,31 @@ const std::string kConsumerProjecId = "X-Endpoint-API-Project-ID";
 
 void CheckServiceControl(std::shared_ptr<context::RequestContext> context,
                          std::function<void(Status status)> continuation) {
-  std::shared_ptr<cloud_trace::CloudTraceSpan> trace_span(
-      CreateSpan(context->cloud_trace(), "CheckServiceControl"));
   // If the method is not configured from the service config.
   // or if not need to check service control, skip it.
   if (!context->method()) {
     if (context->GetRequestHTTPMethodWithOverride() == "OPTIONS") {
-      TRACE(trace_span) << "OPTIONS request is rejected";
       continuation(Status(Code::PERMISSION_DENIED,
                           "The service does not allow CORS traffic.",
                           Status::SERVICE_CONTROL));
     } else {
-      TRACE(trace_span) << "Method is not configured in the service config";
       continuation(Status(Code::NOT_FOUND, "Method does not exist.",
                           Status::SERVICE_CONTROL));
     }
     return;
   } else if (!context->service_context()->service_control() ||
              context->method()->skip_service_control()) {
-    TRACE(trace_span) << "Service control check is not needed";
```

Comment:
I would be careful with this. This changes behavior that is publicly documented: https://cloud.google.com/endpoints/docs/openapi/tracing#spans_created_by_esp

> At a minimum, ESP creates 4 spans per trace

We'll have to update this documentation, not sure if it will confuse pre-existing users to see spans suddenly disappear.

Reply:
On second though, less change, the better.  So I will keep the old spans as they are.